### ISSUE TITLE
cargo udeps panic

### ISSUE THREAD

**Issue by Sherlock-Holo** [est31/cargo-udeps#62]
when I run `cargo udeps` for my project, the udeps panic, it panic at

```
errorthread '<unnamed>' panicked at 'invalid format for extern arg: proc_macro', /home/sherlock/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/cargo-udeps-0.1.9/src/lib.rs:580:25: worker panicked
```

![image](https://user-images.githubusercontent.com/10096425/76677353-3862ec00-6608-11ea-800e-0bc91e4504cb.png)

however I run `cargo check`, it runs successfully.

**Comment by est31** [comment]
Do you have a minimally reproducible example?

**Comment by Sherlock-Holo** [comment]
my project is a littile big, [link](https://github.com/Sherlock-Holo/rfs/commit/9dfb46f3c3fe1ce741f2f2e56c95ebf49b3f9827)

I will try to find a minimally reproducible example, and you can download my project to find out why panic. Sorry for I can't find a minimally reproducible example at first time

**Comment by est31** [comment]
Nevermind I have found an example. It seems to occur in cargo-udeps 0.1.9 first. Thanks for the report.

To reproduce one needs to clone https://github.com/dtolnay/proc-macro-hack and then run `cargo +nightly udeps -p demo-hack` in it.

**Comment by qryxip** [comment]
Probably because of [this](https://blog.rust-lang.org/2020/03/12/Rust-1.42.html#use-proc_macro:tokenstream;-now-works).

> this change will mean that you can delete the `extern crate proc_macro;` line

```console
$ rustc --help -v | rg -- --extern
        --extern NAME[=PATH]
$ touch src/lib.rs && cargo +nightly check -v
       Fresh unicode-xid v0.2.0
       Fresh proc-macro2 v1.0.9
       Fresh quote v1.0.3
       Fresh syn v1.0.16
    Checking proc-macro-hack v0.5.11 (/home/ryo/src/github.com/dtolnay/proc-macro-hack)
     Running `rustc --crate-name proc_macro_hack --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type proc-macro --emit=dep-info,metadata -C prefer-dynamic -C debuginfo=2 -C metadata=60734e183005d2f3 -C extra-filename=-60734e183005d2f3 --out-dir /home/ryo/src/github.com/dtolnay/proc-macro-hack/target/debug/deps -C incremental=/home/ryo/src/github.com/dtolnay/proc-macro-hack/target/debug/incremental -L dependency=/home/ryo/src/github.com/dtolnay/proc-macro-hack/target/debug/deps --extern proc_macro2=/home/ryo/src/github.com/dtolnay/proc-macro-hack/target/debug/deps/libproc_macro2-4610c0a0f83155c0.rmeta --extern quote=/home/ryo/src/github.com/dtolnay/proc-macro-hack/target/debug/deps/libquote-68ac4f6b444e17b8.rmeta --extern syn=/home/ryo/src/github.com/dtolnay/proc-macro-hack/target/debug/deps/libsyn-5ffea411d71f0934.rmeta --extern proc_macro`
    Finished dev [unoptimized + debuginfo] target(s) in 0.21s
```

<https://github.com/est31/cargo-udeps/blob/80b10606a5440ac4e85061b9aedb47e025ad7faa/src/lib.rs#L576-L581>

**Comment by est31** [comment]
Thanks @qryxip for the quick fix. I've released 0.1.10 because of the large impact of the bug.
